include/libtorrent/aux_/session_impl.hpp

```diff
@@ -899,6 +942,12 @@ namespace libtorrent
 			// socket fails.
 			std::vector<std::string> m_outgoing_interfaces;
 
+			// sockets used for outoing utp connections
+			// sockets are created for each device name or IP address in
+			// m_outgoing_interfaces
+			// they are used in a round-robin fashion
+			std::vector<std::shared_ptr<outgoing_udp_socket>> m_outgoing_sockets;
```

Comment:
do you think it would make sense to wrap this, along with the round-robin indices into its own class, and have that as a member here? (I've been trying to think of ways to make session_impl have fewer lines of code). It should also make it easier to unit test.